### DISCUSSION TITLE
Records Not working with RegisterBeanMapper

### DISCUSSION THREAD

**Discussion by krishnakittu** [dropwizard/dropwizard#6164] [General]
Hi Team,

Recently I started DropWizard with JAVA 17,

When I was using Java records as @RegisterBeanMapper its not working whereas with Class its working

// Not Working Scenario
` @SqlQuery("""
            SELECT * from account WHERE name = :name
            """)
    @RegisterBeanMapper(Account.class)
    Optional<Account> getAccoutByName(@Bind("name") String name);
public record Account(String name) {
}`

// Working Scenario

`@SqlQuery("""
            SELECT * from account WHERE name = :name
            """)
    @RegisterBeanMapper(AccountClass.class)
    Optional<AccountClass> getAccoutByName(@Bind("name") String name);`

`
public class AccountClass {

    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
`

**Comment by zUniQueX** [comment]
Hi @krishnakittu. This seems to be an issue with JDBI and not with Dropwizard itself. There is an [open issue](https://github.com/jdbi/jdbi/issues/1822) at JDBI regarding records. There seems to be a solution using `@RegisterConstructorMapper`. Note, that you'll probably have to [use `@ColumnName` or `@ConstructorProperties` annotations](https://jdbi.org/#_constructormapper).
